pkg/catalog/v1alpha1/kubernetes.go

```diff
@@ -31,17 +31,24 @@ import (
 // kubeclient enables kubernetes API operations on catalog
 // instance
 type kubeclient struct {
-	client.Client  // kubernetes client
-	kclient.Handle // manage kubernetes client & enable mocking
+	client.Client   // kubernetes client
+	*kclient.Handle // manage kubernetes client & enable mocking
 }
 
 // kubeclientBuildOption defines the abstraction to build
 // a kubeclient instance
 type kubeclientBuildOption func(*kubeclient)
 
 func (k *kubeclient) withDefaults() error {
-	if k.Client == nil {
-		cli, err := kclient.New()
+	if k.Handle == nil {
```

Comment:
I think we are having typed functions for these assignments as per idiomatic maya.

Reply:
we can have what do you think @AmitKumarDas  ?